src/utils/hooks/useYoutube.js

```diff
@@ -1,52 +1,49 @@
-import { useState, useEffect } from 'react';
+import { useEffect } from 'react';
 import axios from 'axios';
 import videosMock from '../../data/youtube-videos-mock.json';
+import { useMainContext } from '../../state/MainProvider';
 
 function useYoutube(searchTerm = 'wizeline') {
-  const [videos, setVideos] = useState(null);
-  const [search, setSearch] = useState('wizeline');
+  const { state, dispatch } = useMainContext();
```

Comment:
I recommend you to keep your hooks agnostic from the app state. The hooks live on their own, and you might encounter problems if the hooks are used outside the providers.